common/src/main/java/com/revenuecat/purchases/common/caching/DeviceCache.kt

```diff
@@ -96,9 +102,15 @@ open class DeviceCache(
             ?.let { json ->
                 try {
                     val cachedJSONObject = JSONObject(json)
-                    val schemaVersion = cachedJSONObject.optInt("schema_version")
+                    val schemaVersion = cachedJSONObject.optInt(CUSTOMER_INFO_SCHEMA_VERSION_KEY)
+                    val verificationResultString = if (cachedJSONObject.has(CUSTOMER_INFO_VERIFICATION_RESULT_KEY)) {
+                        cachedJSONObject.getString(CUSTOMER_INFO_VERIFICATION_RESULT_KEY)
+                    } else VerificationResult.NOT_VERIFIED.name
+                    cachedJSONObject.remove(CUSTOMER_INFO_SCHEMA_VERSION_KEY)
+                    cachedJSONObject.remove(CUSTOMER_INFO_VERIFICATION_RESULT_KEY)
```

Comment:
Note how I'm removing the extra fields that are added during caching. This is to avoid having the underlying json have extra information it doesn't need when caching it. Before, we were adding the `schema_version` field to the underlying json after caching it, so it wasn't technically equal.